pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -259,64 +294,75 @@ func (c *Controller) syncHandler(key string) error {
 		return err
 	}
 
-	//deploymentName := etcdstorage.Spec.DeploymentName
-	//if deploymentName == "" {
-	//	// We choose to absorb the error here as the worker would requeue the
-	//	// resource otherwise. Instead, the next time the resource is updated
-	//	// the resource will be queued again.
-	//	runtime.HandleError(fmt.Errorf("%s: deployment name must be specified", key))
-	//	return nil
-	//}
-	//
-	//// Get the deployment with the name specified in EtcdStorage.spec
-	//deployment, err := c.deploymentsLister.Deployments(etcdstorage.Namespace).Get(deploymentName)
-	//// If the resource doesn't exist, we'll create it
-	//if errors.IsNotFound(err) {
-	//	deployment, err = c.kubeclientset.AppsV1().Deployments(etcdstorage.Namespace).Create(newDeployment(etcdstorage))
-	//}
-	//
-	//// If an error occurs during Get/Create, we'll requeue the item so we can
-	//// attempt processing again later. This could have been caused by a
-	//// temporary network failure, or any other transient reason.
-	//if err != nil {
-	//	return err
-	//}
-	//
-	//// If the Deployment is not controlled by this EtcdStorage resource, we should log
-	//// a warning to the event recorder and ret
-	//if !metav1.IsControlledBy(deployment, etcdstorage) {
-	//	msg := fmt.Sprintf(MessageResourceExists, deployment.Name)
-	//	c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceExists, msg)
-	//	return fmt.Errorf(msg)
-	//}
-	//
-	//// If this number of the replicas on the EtcdStorage resource is specified, and the
-	//// number does not equal the current desired replicas on the Deployment, we
-	//// should update the Deployment resource.
-	//if etcdstorage.Spec.Replicas != nil && *etcdstorage.Spec.Replicas != *deployment.Spec.Replicas {
-	//	glog.V(4).Infof("EtcdStorage %s replicas: %d, deployment replicas: %d", name, *etcdstorage.Spec.Replicas, *deployment.Spec.Replicas)
-	//	deployment, err = c.kubeclientset.AppsV1().Deployments(etcdstorage.Namespace).Update(newDeployment(etcdstorage))
-	//}
-	//
-	//// If an error occurs during Update, we'll requeue the item so we can
-	//// attempt processing again later. THis could have been caused by a
-	//// temporary network failure, or any other transient reason.
-	//if err != nil {
-	//	return err
-	//}
-
-	//// Finally, we update the status block of the EtcdStorage resource to reflect the
-	//// current state of the world
-	//err = c.updateEtcdStorageStatus(etcdstorage, deployment)
-	//if err != nil {
-	//	return err
-	//}
+	replicasetName := fmt.Sprintf("etcd-rs-%s", etcdstorage.ObjectMeta.Name)
```

Comment:
> I think ReplicaSet name is limited to 253 characters, so we're good to go, or I'm mistaken?

if so, that's great.

Reply:
@deads2k The [Names section of documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) contains some more details about this:

> By convention, the names of Kubernetes resources should be up to maximum length of 253 characters and consist of lower case alphanumeric characters, -, and ., but certain resources have more specific restrictions.

However, I can't find are there any other limitation depending on resource type. Let's give it a try, we can always make it shorter if we run into problems.

Reply:
> However, I can't find are there any other limitation depending on resource type. Let's give it a try, we can always make it shorter if we run into problems.

until you ship it.  Then the names are pinned just about forever.  Otherwise you have a data migration problem.